Project-Euler/Problem017.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Problem 17 - Number letter counts
+ *
+ * @see {@link https://projecteuler.net/problem=17}
+ *
+ * If the numbers 1 to 5 are written out in words: one, two, three, four, five, then there are 3 + 3 + 5 + 4 + 4 = 19 letters used in total.
+ * If all the numbers from 1 to 1000 (one thousand) inclusive were written out in words, how many letters would be used?
+ *
+ * @author Chetan07j
+ */
+
+/**
+ * Function to convert number to word
+ *
+ * This function is called recursively to handle thousand and its sub part
+ */
+const numberToWord = (n) => {
+  let inWord = ''
```

Comment:
Why are you building a word here? A word length would suffice entirely. You could map all their words to their lengths (also dragging this out of the function) and then summing up the lengths rather than concatenating strings.